source/slang/compiler.cpp

```diff
@@ -329,34 +331,32 @@ namespace Slang
             assert(glslang_compile);
         }
 
-        StringBuilder diagnosticBuilder;
-        StringBuilder outputBuilder;
+        List<uint8_t> diagnosticOutput;
+        List<uint8_t> output;
 
-        auto outputFunc = [](char const* text, void* userData)
+        auto outputFunc = [](void const* data, size_t size, void* userData)
         {
-            *(StringBuilder*)userData << text;
+            ((List<uint8_t>*)userData)->AddRange((uint8_t*)data, size);
         };
 
         glslang_CompileRequest request;
         request.sourcePath = "slang";
         request.sourceText = rawGLSL.begin();
-        request.slangStage = (SlangStage) entryPoint->profile.GetStage();
+        request.slangStage = (SlangStage)entryPoint->profile.GetStage();
+        request.disassembleResult = spirvAssembly;
 
         request.diagnosticFunc = outputFunc;
-        request.diagnosticUserData = &diagnosticBuilder;
+        request.diagnosticUserData = &diagnosticOutput;
 
         request.outputFunc = outputFunc;
-        request.outputUserData = &outputBuilder;
+        request.outputUserData = &output;
 
         int err = glslang_compile(&request);
 
-        String diagnostics = diagnosticBuilder.ProduceString();
-        String output = outputBuilder.ProduceString();
```

Comment:
Maybe I'm missing it, but I don't see where any of this code is appending a null byte in the case where text output is desired. Now that we are using a raw-data buffer, we need to guarantee null termination for textual output.

Oh, and on that topic, we *may* have to keep track of the size of the output distinct from `outputSource.Count()` because for textual output you *don't* want the null byte to count as part of the size, right?